nautobot/extras/utils.py

```diff
@@ -103,6 +104,14 @@ def list_subclasses(self):
         return [_class for _class in apps.get_models() if hasattr(_class, "to_objectchange")]
 
 
+@lru_cache(maxsize=5)
```

Comment:
Why 5? Surely there are more change-logged content types than 5? Unless I misunderstand the parameter.

Reply:
This is the max number of distinct cache entries to store, which normally would be keyed/indexed by the parameters passed when the function is called (i.e. so we'd have a cache entry for `my_function(1)` calls that's distinct for the cache entry for `my_function(2)`).

Since this function doesn't actually take any parameters, and always should return the same queryset for the lifetime of a Nautobot process, we could *probably* just set `maxsize=1` but for whatever reason `5` is the value we've used in some other places in the code that should also pretty much always be called in a single way, so I was following that pattern.

Reply:
I would propose just using [`functools.cache`](https://docs.python.org/3/library/functools.html#functools.cache) here since we don't ever want this to expire.

Reply:
I was going that way initially but `functools.cache` is only in python 3.9 and later and we still support 3.8 for now.

Reply:
Oh good catch. I think I saw that and didn't register that we still supported 3.8. Can we do `maxsize=None` then?

Reply:
If I'm reading the docs right, `maxsize=None` actually means "the cache can grow arbitrarily large", but it does say that it's slightly faster than an lru_cache with a maxsize set. I guess since we're only expecting a single cache entry in any case, it's probably fine?

Reply:
I don't think it's possible for the cache to be any larger than 1 entry here, since if you try to pass args or kwargs to this method it will raise an exception.